packages/mongodb-memory-server-core/src/util/getos/index.ts

```diff
@@ -71,151 +74,118 @@ async function getLinuxInfomation(): Promise<LinuxOS> {
   if (envToBool(resolveConfig(ResolveConfigVariables.USE_LINUX_LSB_RELEASE))) {
     log('Forced LSB-Release file!');
 
-    return (await tryLSBRelease()) as LinuxOS;
+    return (await getLSBRelease()) as LinuxOS;
   }
   // Force /etc/os-release to be used
   if (envToBool(resolveConfig(ResolveConfigVariables.USE_LINUX_OS_RELEASE))) {
     log('Forced OS-Release file!');
 
-    return (await tryOSRelease()) as LinuxOS;
+    return (await getOSRelease()) as LinuxOS;
   }
   // Force the first /etc/*-release file to be used
   if (envToBool(resolveConfig(ResolveConfigVariables.USE_LINUX_ANY_RELEASE))) {
     log('Forced First *-Release file!');
 
-    return (await tryFirstReleaseFile()) as LinuxOS;
+    return (await getFirstReleaseFile()) as LinuxOS;
   }
 
   // Try everything
   // Note: these values are stored, because this code should not use "inline value assignment"
 
   log('Trying LSB-Release');
-  const lsbOut = await tryLSBRelease();
+  const lsbOut = await getLSBRelease().catch(() => undefined);
+  log('Trying OS-Release');
+  const osOut = await getOSRelease().catch(() => undefined);
 
   if (!isNullOrUndefined(lsbOut)) {
-    return lsbOut;
+    // add id_like info if available
+    return { ...lsbOut, id_like: osOut?.id_like };
   }
 
-  log('Trying OS-Release');
-  const osOut = await tryOSRelease();
-
   if (!isNullOrUndefined(osOut)) {
     return osOut;
   }
 
   log('Trying First *-Release file');
-  const releaseOut = await tryFirstReleaseFile();
+  const releaseOut = await getFirstReleaseFile().catch(() => undefined);
 
   if (!isNullOrUndefined(releaseOut)) {
     return releaseOut;
   }
 
-  log('Couldnt find an release file');
+  log("Couldn't find a release file");
 
-  // if none has worked, return unkown
+  // if none has worked, return unknown
   return {
     os: 'linux',
-    dist: 'unkown',
+    dist: 'unknown',
     release: '',
   };
 }
 
 /**
  * Try the "lsb_release" command, and if it works, parse it
  */
-async function tryLSBRelease(): Promise<LinuxOS | undefined> {
-  try {
-    const lsb = await promisify(exec)('lsb_release -a'); // exec this for safety, because "/etc/lsb-release" could be changed to another file
-
-    return parseLSB(lsb.stdout);
-  } catch (err) {
-    // check if "USE_LINUX_LSB_RELEASE" is unset, when yes - just return to start the next try
-    if (isNullOrUndefined(resolveConfig(ResolveConfigVariables.USE_LINUX_LSB_RELEASE))) {
```

Comment:
as I see it, it is safe to return a function/promise here instead of catching stuff etc.

the way this currently works is:
if an env var forces one of the methods, that method gets wrapped in a try/catch and then some more spaghetti checks if forcing was enabled and if so, throws the original error. otherwise returns void.

the way this now would work:
if an env var forces one of the methods, that methods gets return and it may throw if it fails.
in the only place where we want to suppress errors, when collecting the info from different sources and take one that worked, we can suppress it as we please.

Reply:
i think of removing these variables, i once thought they could be useful, but from my knowledge it just adds code clutter and didnt provide anything useful - will remove it in some 7.0 beta